src/utilities/focus.ts

```diff
@@ -158,6 +159,61 @@ export function focusLastKeyboardFocusableNode(
   return false;
 }
 
+function getMenuFocusableDescendants(element: HTMLElement) {
+  return element.querySelectorAll(
+    MENUITEM_FOCUSABLE_SELECTORS,
+  ) as NodeListOf<HTMLElement>;
+}
+
+function getCurrentFocusedElementIndex(
+  allFocusableChildren: NodeListOf<HTMLElement>,
+  currentFocusedElement: HTMLElement,
+) {
+  let currentItemIdx = 0;
+
+  for (const focusableChild of allFocusableChildren) {
+    if (focusableChild === currentFocusedElement) {
+      break;
+    }
+    currentItemIdx++;
+  }
+  return currentItemIdx;
```

Comment:
(request) To mimic [Array.prototype.findIndex()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) we should return `-1` if `currentFocusedElement` is not found in `allFocusableChildren`

```suggestion
  return currentItemIdx === allFocusableChildren.length ? -1 : currentItemIdx;
```

Reply:
Smart! Thank you!